c++/src/kj/CMakeLists.txt

```diff
@@ -163,6 +163,29 @@ if(NOT CAPNP_LITE)
   install(FILES ${kj-http_headers} DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/kj/compat")
 endif()
 
+# kj-tls ======================================================================
+set(kj-tls_sources
+  compat/readiness-io.c++
+  compat/tls.c++
+)
+set(kj-tls_headers
+  compat/readiness-io.h
+  compat/tls.h
+)
+if(NOT CAPNP_LITE)
+  add_library(kj-tls ${kj-tls_sources})
+  add_library(CapnProto::kj-tls ALIAS kj-tls)
+  target_link_libraries(kj-tls PUBLIC kj-async kj)
```

Comment:
```suggestion
  target_link_libraries(kj-tls PUBLIC kj-async PRIVATE kj)
```

Reply:
@Sarcasm why did you remove `kj`?
Even though `kj` is public in `kj-async`, `kj` is still a direct dependency of `kj-tls`, therefore it should be explicitly added in `target_link_libraries` (a target shouldn't rely on public dependencies of its dependencies, if there are also direct dependencies).

Reply:
> a target shouldn't rely on public dependencies of its dependencies

Why shouldn't it?

When the code is under control, here a kj lib depending on another kj lib, I don't see how that is a problem.
For me kj-async is kj + async, kj is implied.

Will CMake generate functionally different compile or link commands?

I'm okay to change, but ` target_link_libraries(kj-tls PUBLIC kj-async PRIVATE kj)` feels like being over specifying to me.

Thanks for the review by the way!

Reply:
You're right, it doesn't change generated compiler and linker commands, but il will break if `kj` is removed from INTERFACE property of `kj-async`, unless `kj` is added as a dependency of `kj-async`. `kj-tls` should'nt care of `kj-async` properties.
Moreover, removing `kj` makes dependencies between targets harder to understand (because if you do this for this target, maybe others targets follow this practice, therefore maintenairs will be suspicous about CMakeLists).

Reply:
We just have to know what is part of the public API.
kj-async declaring kj as a PUBLIC or INTERFACE dependency is a contract with kj users.
Changing that would be an API break.
And if that changes in the headers, it can also be changed in the CMakeLists.txt.

To make a parallel with the STL.
When I use `std::map`, I don't include `utility` for `std::pair` manually, because I assume it's part of the public API of `std::map`.
There is nothing suspicious about that, we just have to find a middle ground.

There is no point in discussing this too much, it might be best to the kj maintainers decide what they want for the project.

Reply:
In the past, I've operated under the assumption that `kj-*` CMake targets will always publicly depend on `kj`, and `capnp-*` targets will always publicly depend on `capnp`. For example, in samples/CMakeLists.txt, we just link with `CapnProto::capnp-rpc`. So my personal style would have been not to list the explicit `kj` dependency.

That does go against an "include what you use" philosophy, though, which I also like. I would say for the purposes of merging, either one is just fine.